src/Google.Api.Gax/Platform.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Threading;
+using System.Net;
+
+namespace Google.Api.Gax
+{
+    public enum PlatformType
+    {
+        Unknown = 0,
+        Gce = 1,
+        Gae = 2,
+    }
+
+    public sealed class GcePlatformDetails
+    {
+        public GcePlatformDetails(string metadataJson)
+        {
+            MetadataJson = metadataJson;
+            var json = Newtonsoft.Json.Linq.JObject.Parse(metadataJson);
+            ProjectId = json["project"]["projectId"].ToString();
+            InstanceId = json["instance"]["id"].ToString();
+            ZoneName = json["instance"]["zone"].ToString();
+        }
+
+        public string MetadataJson { get; }
+        public string ProjectId { get; }
+        public string InstanceId { get; }
+        public string ZoneName { get; }
+    }
+
+    public sealed class GaePlatformDetails
+    {
+        public GaePlatformDetails(string gcloudProject, string gaeInstance, string gaeVersion)
+        {
+            ProjectId = gcloudProject;
+
```

Comment:
Are the other two properties supposed to be set here from the unused parameters?